R/run-tests.sh

```diff
@@ -30,11 +30,22 @@ if [[ $(echo $SPARK_AVRO_JAR_PATH | wc -l) -eq 1 ]]; then
 fi
 
 if [ -z "$SPARK_JARS" ]; then
-  SPARK_TESTING=1 NOT_CRAN=true $FWDIR/../bin/spark-submit --driver-java-options "-Dlog4j.configuration=file:$FWDIR/log4j.properties" --conf spark.hadoop.fs.defaultFS="file:///" --conf spark.driver.extraJavaOptions="-Dio.netty.tryReflectionSetAccessible=true" --conf spark.executor.extraJavaOptions="-Dio.netty.tryReflectionSetAccessible=true" $FWDIR/pkg/tests/run-all.R 2>&1 | tee -a $LOGFILE
+  JARS=""
 else
-  SPARK_TESTING=1 NOT_CRAN=true $FWDIR/../bin/spark-submit --jars $SPARK_JARS --driver-java-options "-Dlog4j.configuration=file:$FWDIR/log4j.properties" --conf spark.hadoop.fs.defaultFS="file:///" --conf spark.driver.extraJavaOptions="-Dio.netty.tryReflectionSetAccessible=true" --conf spark.executor.extraJavaOptions="-Dio.netty.tryReflectionSetAccessible=true" $FWDIR/pkg/tests/run-all.R 2>&1 | tee -a $LOGFILE
+  JARS="--jars $SPARK_JARS"
 fi
 
+NOT_CRAN=true \
+SPARK_TESTING=1 \
+$FWDIR/../bin/spark-submit \
+--conf spark.driver.extraJavaOptions="-Dio.netty.tryReflectionSetAccessible=true" \
+--conf spark.executor.extraJavaOptions="-Dio.netty.tryReflectionSetAccessible=true" \
+--conf spark.hadoop.fs.defaultFS="file:///" \
+--conf spark.r.daemonInit='testInit <- "wow"' \
+--driver-java-options "-Dlog4j.configuration=file:$FWDIR/log4j.properties" \
+$JARS \
+$FWDIR/pkg/tests/run-all.R 2>&1 | tee -a $LOGFILE
```

Comment:
The only thing new here is "spark.r.daemonInit" -- the rest is just a refactoring of a duplication above.